### DISCUSSION TITLE
Building sirius-web locally is not working

### DISCUSSION THREAD

**Discussion by littlenici** [eclipse-sirius/sirius-components#3160] [General]
Hi,
I'm trying to build sirius-web locally since thursday. Unfortunately it runs in an error with a HttpResponseException:
[sirius-web_backend.log](https://github.com/eclipse-sirius/sirius-web/files/14402717/sirius-web_backend.log)

In the attached log you can see the HttpResponseException: status code: 401, reason phrase: Unauthorized (401).
This is because the dependent URLs for example: https://maven.pkg.github.com/eclipse-sirius/emf-json , the references are not available under the url. All URLs are not working.
I have tried to build the dependet project afterwards on my own - also locally. 
I cloned the https://github.com/eclipse-sirius/sirius-emf-json from the repo, than I took the instalation guide from the repo but the project as well runs in build errors. 
In a second step I have tried to resolve the Flow-Designer dependency to build the jar file. 
I cloned the project: https://github.com/ObeoNetwork/Flow-Designer and also checked the installation guidiance unter the project in github. I have tried to resolve the dependeny errors from that project, never the less it led again to an enourmous amount of dependency errors, where I would need more than one week to resolve them. This makes no sense. 
Afterwards I have tried to build the papyrus-web: https://gitlab.eclipse.org/eclipse/papyrus/org.eclipse.papyrus-web#build
Here as well I got the same erros like I had with sirius-web. 
I cannot got on without your help to build it.
How can I resolve the dependencies to:
-     sirius-emf-json
-     github-sirius-components
-     Flow-Designer
-     github-papyrus-sirius-uml2
Thank you in advance and best regards
Nicole

**Comment by AxelRICHARD** [comment]
Hello,
It seems you didn't follow the build instructions about the github token: https://github.com/eclipse-sirius/sirius-web?tab=readme-ov-file#github-access-token
Hope it will help you.
Regards,

**Reply by littlenici** [reply]
Would be great if this would resolve my problem. Unfortunately I did all the guidiance steps, but it's not working. Could you please help me to resolve the dependent projects?

**Reply by AxelRICHARD** [reply]
The `Unauthorized (401)` error on github maven/node packages are typically related to the github token.
Could you verify again the following points please?
- Did you generate a token with the appropriate rights?
- Do you use it in your local `settings.xml`?
- Do you use it in your local `.npmrc`?
- Do you use the appropriate versions of Java and Maven as mentioned [here](https://github.com/eclipse-sirius/sirius-web?tab=readme-ov-file#requirements)?

**Reply by littlenici** [reply]
Hi. thanks for helping me so fast, It was the .npmrc, it took a while to found out, why it wasn't working. Now I built it! Thanks a lot!

**Comment by baristezel** [comment]
Hi,

Me and one of my colleagues have tried to build sirius web almost a week. However, we stuck on final step. While maven tries to build "sirius-web-sample-application", it gives always same build failure.  We verified all issue about github access token such as settining settings.xml and .npmrc. But we could't find any problem on these files. You may see error at below of the post.


<pre>[<font color="#12488B"><b>INFO</b></font>] <font color="#C01C28"><b>BUILD FAILURE</b></font>
[<font color="#12488B"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#12488B"><b>INFO</b></font>] Total time:  04:55 min
[<font color="#12488B"><b>INFO</b></font>] Finished at: 2024-03-23T12:17:35+01:00
[<font color="#12488B"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#C01C28"><b>ERROR</b></font>] Failed to execute goal on project <font color="#2AA1B3">sirius-web-sample-application</font>: <font color="#C01C28"><b>Could not resolve dependencies for project org.eclipse.sirius:sirius-web-sample-application:jar:2024.3.0: Failed to collect dependencies at BPMN2:org.obeonetwork.dsl.bpmn2:jar:4.0.3-SNAPSHOT</b></font>: Failed to read artifact descriptor for BPMN2:org.obeonetwork.dsl.bpmn2:jar:4.0.3-SNAPSHOT: Could not transfer artifact BPMN2:org.obeonetwork.dsl.bpmn2:pom:4.0.3-SNAPSHOT from/to github-bpmn (https://maven.pkg.github.com/ObeoNetwork/BPMN-Designer): Authentication failed for https://maven.pkg.github.com/ObeoNetwork/BPMN-Designer/BPMN2/org.obeonetwork.dsl.bpmn2/4.0.3-SNAPSHOT/org.obeonetwork.dsl.bpmn2-4.0.3-SNAPSHOT.pom 401 Unauthorized -&gt; <b>[Help 1]</b>
[<font color="#C01C28"><b>ERROR</b></font>] 
[<font color="#C01C28"><b>ERROR</b></font>] To see the full stack trace of the errors, re-run Maven with the <b>-e</b> switch.
[<font color="#C01C28"><b>ERROR</b></font>] Re-run Maven using the <b>-X</b> switch to enable full debug logging.
[<font color="#C01C28"><b>ERROR</b></font>] 
[<font color="#C01C28"><b>ERROR</b></font>] For more information about the errors and possible solutions, please read the following articles:
[<font color="#C01C28"><b>ERROR</b></font>] <b>[Help 1]</b> http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[<font color="#C01C28"><b>ERROR</b></font>] 
[<font color="#C01C28"><b>ERROR</b></font>] After correcting the problems, you can resume the build with the command
[<font color="#C01C28"><b>ERROR</b></font>]   <b>mvn &lt;args&gt; -rf :sirius-web-sample-application</b>
</pre>

Thanks for your advice.
Baris

**Comment by hakut** [comment]
Hi,

I'm also trying to build but without success.
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Java version: 17.0.10, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-17
```
I created an access token as suggested (read:package) and added that token to my .m2/settings.xml and also ~/.npmrc
However when I run `mvn clean install -f packages/pom.xml -DskipTests` it gives me following error:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for sirius-components 2024.3.1:
[INFO]
[INFO] sirius-web-sample-application ...................... FAILURE [01:09 min]
[INFO] sirius-web-domain .................................. SKIPPED
[INFO] sirius-web-application ............................. SKIPPED
[INFO] sirius-web-papaya .................................. SKIPPED
[INFO] sirius-web-infrastructure .......................... SKIPPED
[INFO] sirius-web-starter ................................. SKIPPED
[INFO] sirius-web-tests ................................... SKIPPED
[INFO] sirius-web ......................................... SKIPPED
[INFO] sirius-web-parent .................................. SKIPPED
[INFO] sirius-components-test-coverage-aggregation ........ SKIPPED
[INFO] sirius-components-releng-parent .................... SKIPPED
[INFO] sirius-components .................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:11 min
[INFO] Finished at: 2024-04-13T23:55:19+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project sirius-web-sample-application: Could not resolve dependencies for project org.eclipse.sirius:sirius-web-sample-application:jar:2024.3.1: Failed to collect dependencies at BPMN2:org.obeonetwork.dsl.bpmn2:jar:4.0.3-SNAPSHOT: Failed to read artifact descriptor for BPMN2:org.obeonetwork.dsl.bpmn2:jar:4.0.3-SNAPSHOT: Could not transfer artifact BPMN2:org.obeonetwork.dsl.bpmn2:pom:4.0.3-SNAPSHOT from/to github-bpmn (https://maven.pkg.github.com/ObeoNetwork/BPMN-Designer): Authentication failed for https://maven.pkg.github.com/ObeoNetwork/BPMN-Designer/BPMN2/org.obeonetwork.dsl.bpmn2/4.0.3-SNAPSHOT/org.obeonetwork.dsl.bpmn2-4.0.3-SNAPSHOT.pom 401 Unauthorized -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```
Thank you in advance.
Samet

**Reply by hakut** [reply]
I don't know what is the real reason but I tried to build it using **cmd** instead of **powershell** and it worked. If it helps to anyone somehow, I'm more than happy.
